public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/SimpleCondition.tsx

```diff
@@ -20,7 +20,7 @@ import { updateExpression } from './reducer';
 export interface SimpleCondition {
   whenField?: string;
   evaluator: {
-    params: number[];
+    params: Array<number | string | undefined>;
```

Comment:
Why we need to allow `string` and `undefined`?

Reply:
Because the code handles the case of typing a negative number:  when a user is typing a negative number, they first type the '-' character. During this intermediate state, the value is a string ('-') before it becomes a complete negative number.